signalfx-tracing/signalfx-python-tracing/django/client.py

```diff
@@ -0,0 +1,111 @@
+#
+# #Then in your main function, at the end of all function calls to be traced:
+# tracer.close()
+#
+# These are noted below with additional explanations.
+
+
+import logging
+import requests
+from requests.auth import HTTPBasicAuth
+from signalfx_tracing import auto_instrument, create_tracer, trace
+# Important: Needed to propagate traces
+from signalfx_tracing.libraries import requests_config
+
+
+# Create and configure logger
+logging.basicConfig(level='DEBUG',
+                    format='%(asctime)s %(message)s',)
+
+
+# create an instance of the SignalFx tracer
+tracer = create_tracer(service_name='BlogClient')
+
+# Setting requests_config.propagate to 'True'  ensures
+# that the relational information between traces is maintained
+# and forwarded.
+requests_config.propagate = True
+
+# Configure auto-instrumentation of the tracing instance
+auto_instrument(tracer)
+
+
+class BlogClient:
+    def __init__(self, url, username, password):
+        self.url = url
+        self.session = requests.Session()
+        self.session.auth = HTTPBasicAuth(username, password)
+
+    @trace
+    def create_blog(self, new_title, new_description, new_body):
+        response = requests.post(self.url,
+                                 json=dict(title=new_title,
+                                           description=new_description,
+                                           body=new_body),
+                                 auth=self.session.auth, )
+        success = (response.status_code == 201)
+        logging.info('INFO - Created: {}'.format(success))
+        return response.json().get('id')
+
+    @trace
+    def list_blogs(self):
+        response = requests.get(self.url)
+        success = (response.status_code == 200)
+        logging.info('INFO - Viewed: {}'.format(success))
+        return response.json()
+
+    @trace
+    def get_blog(self, blog_id):
+        url_to_read = self.url.format(blog_id)
+        response = requests.get(url_to_read)
+        success = (response.status_code == 200)
+        logging.info('INFO - Read: {}'.format(success))
+        return response.json()
+
+    @trace
+    def update_blog(self, blog_id, new_title):
+        url_to_update = self.url + '{}/'.format(blog_id)
+        response = requests.patch(url_to_update,
+                                  json=dict(title=new_title),
+                                  auth=self.session.auth, )
+        success = (response.status_code == 200)
+        logging.info('INFO - Updated: {}'.format(success))
+        return response.json()
+
+    @trace
+    def delete_blog(self, blog_id):
+        url_to_delete = self.url + '{}/'.format(blog_id)
+        response = requests.delete(url_to_delete,
+                                   auth=self.session.auth, )
+        success = (response.status_code == 204)
+        logging.info('INFO - Deleted status: {}'.format(success))
+
+
+def main():
+    client = BlogClient('http://localhost:8001/articles/', 'admin', 't00c0mm0n')
```

Comment:
Should this remain here?

Reply:
Thanks for spotting that (it slipped when I was re-testing). Updated it to 'USERNAME' and 'PASSWORD'

Reply:
That's fine. Another approach is to make this program take command line arguments with a quick pass through `argparse.ArgumentParser` and use that to initialize the `BlogClient`.

Reply:
Right! Updated `client.py` to take in CLI arguments.